### ISSUE TITLE
Security - UI - Resize Behavior

### ISSUE THREAD

**Issue by Mgranfie** [Apicurio/apicurio-ux-design#53]
When long names are given to Scopes, in both the creation stage (security requirements) and the assignment stage (security schemes), the presentation of the content is not displaying or resizing properly.

<img width="819" alt="screen shot 2018-12-07 at 10 46 49 am" src="https://user-images.githubusercontent.com/18149547/49657639-21e5e200-fa0e-11e8-8ea6-ace11a20a7c3.png">
<img width="911" alt="screen shot 2018-12-07 at 10 46 34 am" src="https://user-images.githubusercontent.com/18149547/49657649-27432c80-fa0e-11e8-8deb-87c76b075294.png">

**Comment by Mgranfie** [comment]
@EricWittmann including the resize suggestions here...

This design is following the PatternFly Table resize behavior
1. Set default size and truncate text to fit and wrap and fit, depending on how many lines per row.
2. Icons are anchored to the right margins and top line of text
3. Set a minimum table size, wrapping and stacking the text

**Default Table**
![scope_default](https://user-images.githubusercontent.com/18149547/49672053-dd6f3c00-fa37-11e8-9a92-44d90c428e88.png)

Resized table with minimum sizes
![scope_resized](https://user-images.githubusercontent.com/18149547/49672071-f11aa280-fa37-11e8-8459-37155cf42b2e.png)

**List of Security Configurations**

This looks like it uses the PatternFly list view and launched this layout from that. Would inherit the list behaviors. 
The only behaviors that were not defined and are not clear are the check all at the top of the list vs. checking an individual line item. Typically the interaction goes something like this:
1. Check All at top of list has a hover text and when selected, all items in the list are selected.
2. In the case that the line item is rolled up, and the user clicks the select all box, we may want to expand the content below to show they have all been selected and that there is more content.
3. If All is selected and the user de-selects one of the line items in the subcontent, the main select all button should be deselected.
4. If the user individually selects all the line items, one at a time, once they have all been selected, the "select all" box should become selected.

<img width="738" alt="configure_security" src="https://user-images.githubusercontent.com/18149547/49672331-b8c79400-fa38-11e8-92a6-9ac11ff519dd.png">

Let me know what your thoughts are and if I am understanding the tasks and interactions correctly.
Thanks,
Monica

**Comment by EricWittmann** [comment]
The mockups of course makes sense.  What's missing from the screenshots but present in your explanatory text is the Check All functionality.  However, I don't think we need that in this use-case.  It is quite unlikely that the user will ever want to select all checkboxes, either for the list of security schemes **or** the list of scopes within a selected scheme.  So I'd suggest we leave that off.

The other concern I have is scaling - in the case that there are a LOT of scopes.  But that's probably pretty unlikely as well.  Showing them in a list is probably the right thing to do.

So this looks good I think.

**Comment by Mgranfie** [comment]
@EricWittmann I would agree with leaving the select all off, unless there are large volumes of information it's not needed. 
The list, although another list : ) the challenge of all data heavy applications... is clean, expected and extensible in a variety of ways, and easily scales for interaction/resize. So it is a logical selection in those ways.
So it sounds like we are good with this?...

**Comment by EricWittmann** [comment]
I think so yes!  Thanks. :)

**Comment by Mgranfie** [comment]
Current resize

Before resize

![Screen Shot 2019-03-19 at 4 03 24 PM](https://user-images.githubusercontent.com/18149547/54638173-c68dac00-4a60-11e9-8cd3-aab0e99672f8.png)


Resized (the fonts get bigger and the name gets cut off...)

![Screen Shot 2019-03-19 at 4 03 06 PM](https://user-images.githubusercontent.com/18149547/54638187-d3aa9b00-4a60-11e9-8601-3637c1ea3af9.png)

**Comment by EricWittmann** [comment]
The fonts getting bigger is a surprise.

**Comment by Mgranfie** [comment]
Oh yikes. I know, funky huh. It's good though I can read it without my
glasses on!! : )
